docarray/array/storage/elastic/seqlike.py

```diff
@@ -58,34 +58,58 @@ def __repr__(self):
         """
         return f'<{self.__class__.__name__} (length={len(self)}) at {id(self)}>'
 
-    def _upload_batch(self, docs: Iterable['Document']):
+    @staticmethod
+    def _parse_index_ids_from_bulk_info(
+        accumulated_info: List[Dict],
+    ) -> Dict[str, List[int]]:
+        """Parse ids from bulk info of failed send request to ES operation
+
+        :param accumulated_info: accumulated info of failed operation
+        :return: dict containing failed index ids of each operation type
+        """
+
+        parsed_ids = {}
+
+        for info in accumulated_info:
+            for _op_type in info.keys():
+                if '_id' in info[_op_type]:
+                    if _op_type not in parsed_ids:
+                        parsed_ids[_op_type] = []
+
+                    parsed_ids[_op_type].append(info[_op_type]['_id'])
+
+        return parsed_ids
+
+    def _upload_batch(self, docs: Iterable['Document']) -> List[int]:
         batch = []
-        failed_index = []
+        accumulated_info = []
         for doc in docs:
             batch.append(self._document_to_elastic(doc))
             if len(batch) > self._config.batch_size:
-                failed_index.extend(self._send_requests(batch))
+                accumulated_info.extend(self._send_requests(batch))
                 self._refresh(self._config.index_name)
                 batch = []
         if len(batch) > 0:
-            failed_index.extend(self._send_requests(batch))
+            accumulated_info.extend(self._send_requests(batch))
             self._refresh(self._config.index_name)
 
-        return failed_index
+        successful_ids = self._parse_index_ids_from_bulk_info(accumulated_info)
+        if 'index' not in successful_ids:
```

Comment:
what else is in the `successful_ids`?

Reply:
Why not extracting directly a list from `parse_index_ids_from_bulk_info`?

Reply:
as @samsja mention in previous thread that the function `_parse_index_ids_from_bulk_info` could be utilized for a more general purpose, I take this as we could extract any ids from any `_op_type` provided from info ( currently we only extract `_op_type` = `index` in this function but if we maybe need to extract `delete` we can utilize this as welll. So I make that the function return like this 

```python
{
    "index" : [1,2,3,4,5] # successfull indexed ids,
    "delete" : [6,7] # successfull deleted index ids
}
```